lib/neovim-plugin.nix

```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  nixvimOptions,
+}:
+with lib; {
+  mkSetupOptionsOption = pluginName: options: {
+    type = with types;
+      submodule {
+        freeformType = with types; attrsOf anything;
+        inherit options;
+      };
+    description = ''
+      Options provided to the `require('${pluginName}').setup` function.
+    '';
+    default = {};
+    example = {
+      foo_bar = 42;
+      hostname = "localhost:8080";
+      callback.__raw = ''
+        function()
+          print('nixvim')
+        end
+      '';
+    };
+  };
+
+  extraOptionsOptions = {
+    extraOptions = mkOption {
+      default = {};
+      type = types.attrs;
+      description = ''
+        These attributes will be added to the table parameter for the setup function.
+        (Can override other attributes set by nixvim)
+      '';
+    };
+  };
```

Comment:
We won't need this in the future right? As the setupOptions are exactly the same?

While we are at it can you switch from attrs to attrsOf anything

Reply:
> We won't need this in the future right? As the setupOptions are exactly the same?

In the future no, we won't need that.
But for backwards compatibility, we have to keep it for now.



> While we are at it can you switch from attrs to attrsOf anything

done !